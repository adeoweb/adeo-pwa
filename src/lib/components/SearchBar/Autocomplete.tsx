import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import PRODUCT_SEARCH from '../../queries/productSearch.graphql';
import { useAutocomplete } from 'src/peregrine/lib/talons/adeoweb/SearchBar/useAutocomplete';
import Suggestions from 'src/lib/components/SearchBar/Suggestions';

type TAutocompleteProps = {
    setVisible: (isVisible: boolean) => void;
    visible: boolean;
    searchValue: string;
};

const MESSAGES = new Map()
    .set('ERROR', 'An error occurred while fetching results.')
    .set('LOADING', 'Fetching results...')
    .set('PROMPT', 'Search for a product')
    .set('EMPTY_RESULT', 'No results were found.');

const Autocomplete: FunctionComponent<TAutocompleteProps> = ({
    setVisible,
    visible,
    searchValue
}) => {
    const { t } = useTranslation();
    const { messageType, products } = useAutocomplete({
        query: PRODUCT_SEARCH,
        visible,
        value: searchValue
    });
    const wrapperClass = `autocomplete-suggestions${visible ? ' show' : ''}`;
    const message = t(MESSAGES.get(messageType));

    return (
        <div className={wrapperClass}>
            {message && <div className="autocomplete-title">{message}</div>}
            <Suggestions
                products={products}
                setVisible={setVisible}
                visible={visible}
            />
        </div>
    );
};

export default Autocomplete;
