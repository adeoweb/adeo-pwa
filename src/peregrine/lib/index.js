// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
