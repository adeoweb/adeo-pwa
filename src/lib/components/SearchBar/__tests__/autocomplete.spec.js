import React from 'react';

import { createTestInstance } from '@magento/peregrine';

import Autocomplete from '../Autocomplete';

jest.mock('src/lib/components/SearchBar/Suggestions', () => () => null);
jest.mock('src/peregrine/lib/talons/adeoweb/SearchBar/useAutocomplete', () => {
    return {
        useAutocomplete: () => ({
            messageType: 'PROMPT',
            products: undefined
        })
    };
});
const setVisible = jest.fn();

test('renders correctly when visible', () => {
    const tree = createTestInstance(
        <Autocomplete setVisible={setVisible} visible={true} searchValue={''} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders correctly when not visible', () => {
    const tree = createTestInstance(
        <Autocomplete
            setVisible={setVisible}
            visible={false}
            searchValue={''}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
