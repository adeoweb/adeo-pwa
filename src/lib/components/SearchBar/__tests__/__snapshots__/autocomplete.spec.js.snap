// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not visible 1`] = `
<div
  className="autocomplete-suggestions"
>
  <div
    className="autocomplete-title"
  >
    Search for a product
  </div>
</div>
`;

exports[`renders correctly when visible 1`] = `
<div
  className="autocomplete-suggestions show"
>
  <div
    className="autocomplete-title"
  >
    Search for a product
  </div>
</div>
`;
